airbyte-api/src/main/openapi/config.yaml

```diff
@@ -3229,6 +3231,54 @@ components:
           type: string
         stats:
           $ref: "#/components/schemas/AttemptStats"
+    AttemptFailureSummary:
+      type: object
+      required:
+        - failures
+      properties:
+        failures:
+          type: array
+          items:
+            $ref: "#/components/schemas/AttemptFailureReason"
+        partialSuccess:
+          description: True if the number of committed records for this attempt was greater than 0. False if 0 records were committed. Blank if number of committed records is unknown.
+          type: boolean
+    AttemptFailureReason:
+      type: object
+      required:
+        - failureOrigin
+        - timestamp
+      properties:
+        failureOrigin:
+          $ref: "#/components/schemas/AttemptFailureOrigin"
+        failureType:
+          $ref: "#/components/schemas/AttemptFailureType"
+        externalMessage:
+          type: string
+        stacktrace:
+          type: string
+        timestamp:
+          type: integer
+          format: int64
+    AttemptFailureOrigin:
+      description: Indicates where the error originated. If not set, the origin of error is not well known.
+      type: string
+      enum:
+        - unknown
+        - source
+        - destination
+        - replication_worker
+        - persistence
+        - normalization
+        - dbt
+    AttemptFailureType:
+      description: Categorizes well known errors into types for programmatic handling. If not set, the type of error is not well known.
+      type: string
+      enum:
+        - unknown
+        - user_error
+        - system_error
+        - transient
```

Comment:
Not sure if transient makes sense since the underlying error is likely a system error or not known.

Reply:
Was wondering the same. Would it maybe make sense to have `transient` (if we're able to determine it) as a property directly on `AttemptFailureReason` instead?

Reply:
I went ahead and removed 'transient' from this 'failureType' enum and added a 'retryable' boolean to the 'AttemptFailureReason' instead. I think 'retryable' can be used to give more explicit direction to users compared to 'transient'